nfp/preprocessing/preprocessor.py

```diff
@@ -159,7 +160,10 @@ def __call__(self,
         Dict[str, np.ndarray]
             A dictionary of key, array pairs as a single sample.
         """
-        nx_graph = self.create_nx_graph(structure, **kwargs)
+        if jsonfile == None:
+            nx_graph = self.create_nx_graph(structure, **kwargs)
+        else:
+            nx_graph = self.create_nx_graph(structure, jsonfile, **kwargs)
```

Comment:
the **kwargs option here was to allow you to pass new inputs in subclasses, without having to edit the code here, so I dont think we need any changes to preprocessor.py to get this to work

Reply:
Done. it is in the new commit